### PR TITLE
automatically fill command

### PR DESCRIPTION
**Pull request by Liaojinghui**
Automatically fill the command when `tap` is clicked.
switch possible commands if more `tap` is clicked.

### CODE REVIEW

**Comment by Liaojinghui** [comment]
input:
image

after tap:
image

click tap again:
image

click tap again:
image

click tap again:
image

**Comment by erikzhang** [comment]
@neo-project/ngd-shanghai Need to test.

**Comment by superboyiii** [comment]
Why we have to add a blank at the beginning of command input? When I delete the blank and input anything, the first character can't be deleted. And I find when I push the left button `<-` on my keyboard, the cursor go to the right.

**Comment by Liaojinghui** [comment]
> When I delete the blank and input anything, the first character can't be deleted. And I find when I push the left button `<-` on my keyboard, the cursor go to the right.

fixed

**Comment by superboyiii** [comment]
> I push the left button `<-` on my keyboard, the cursor go to the right.

when I push the left button <- on my keyboard, the cursor still goes to the right.

File: Neo.ConsoleService/ConsoleServiceBase.cs [hunk 1]
```diff
@@ -601,17 +610,102 @@ public virtual void RunConsole()
...
+                                break;
+                            }
+                        case ConsoleKey.Backspace:
+                            {
+
+                                if (currentCursor > cursorStart)
+                                {
+                                    builder.Remove(currentCursor - cursorStart - 1, 1);
+
+                                    Console.SetCursorPosition(currentCursor - 1, currentLine);
```
Comment (shargon) [inline Neo.ConsoleService/ConsoleServiceBase.cs:658]: Clean this empty lines

Comment (Liaojinghui) [reply Neo.ConsoleService/ConsoleServiceBase.cs:658]: > Clean this empty lines

Hi shargon, did you test this on Linux XD

Comment (shargon) [reply Neo.ConsoleService/ConsoleServiceBase.cs]: I mean the source code empty lines, for example 657

Comment (Liaojinghui) [reply Neo.ConsoleService/ConsoleServiceBase.cs]: I know, i removed those lines as you mentioned. But it looks like this pr has some issue running on Linux,,,,,that is why i ask~~

**Comment by Liaojinghui** [comment]
I've tested this on Windows, MacOS and Ubuntu, and I did not find any issue with it. Just rented a centOS server, will try it later.

**Comment by Liaojinghui** [comment]
Tested on CenOS 8.2. Nothing special happened.

**Comment by cloud8little** [comment]
neo client exit after this command: "invoke 0x49cf4e5378ffcd4dec034fd98a174c5491e395e2  designateAsRole [{"type":"Integer","value":4},{"type":"Array","va"

af7537a27c2de1816e0009a8025f365

https://github.com/neo-project/neo-node/pull/818/files#diff-7db18a91de85fd796cd1cbbefb19ce9d361535852c600bfd35923f86f687217bR692

**Comment by cloud8little** [comment]
Another issue, after designate the node as state service role, it exited with following message.

b4904a725f94e8c0e9bc107db157858

ff3e63dd7118809d022f738ed8ae3f3

**Comment by Liaojinghui** [comment]
image

**Comment by cloud8little** [comment]
https://github.com/neo-project/neo-node/pull/818#issuecomment-938497540 this issue still exists.

**Comment by Liaojinghui** [comment]
You might not believe it, but I think it is a bug of dotnet........

Simply add: 
```C#
var lineStart = Console.CursorTop;
```
Screenshot from 2021-10-14 18-25-02

And you get:

Screenshot from 2021-10-14 18-26-22

**Comment by cloud8little** [comment]
@shargon could you help @Liaojinghui with fixing the UT?

**Comment by Liaojinghui** [comment]
Update the dotnet SDK to 6.0, hope the bug is fixed. will test later.

**Pull request closed without merging**